src/java/org/apache/cassandra/metrics/AccordMetrics.java

```diff
@@ -0,0 +1,279 @@
+     */
+    public final Meter coordinatorTimedoutTrx;
+
+    /**
+     * The time between the start of the transaction and the start of the recovery, if the transaction is recovered.
+     */
+    public final Timer coordinatorRecoveryDelay;
+
+    /**
+     * The time between the start of the recovery and the execution of the transaction, if the transaction is recovered.
+     */
+    public final Timer coordinatorRecoveryDuration;
+
+    /**
+     * The ratio of the number of fast path transactions to the total number of transactions.
+     */
+    public final RatioGaugeSet fastPathToTotal;
+
+    private AccordMetrics(String scope)
+    {
+        DefaultNameFactory replica = new DefaultNameFactory("accord-replica", scope);
+        commitLatency = Metrics.timer(replica.createMetricName("CommitLatency"));
+        executeLatency = Metrics.timer(replica.createMetricName("ExecuteLatency"));
+        applyLatency = Metrics.timer(replica.createMetricName("ApplyLatency"));
+        applyDuration = Metrics.timer(replica.createMetricName("ApplyDuration"));
+        deps = Metrics.histogram(replica.createMetricName("Dependencies"), true);
+        progressLogSize = Metrics.meter(replica.createMetricName("ProgressLogSize"));
+
+        DefaultNameFactory coordinator = new DefaultNameFactory("accord-coordinator", scope);
+        coordinatorDeps = Metrics.histogram(coordinator.createMetricName("Dependencies"), true);
+        coordinatorFastPathTrx = Metrics.meter(coordinator.createMetricName("FastPathTrx"));
+        coordinatorSlowPathTrx = Metrics.meter(coordinator.createMetricName("SlowPathTrx"));
+        coordinatorPreemptedTrx = Metrics.meter(coordinator.createMetricName("PreemptedTrx"));
+        coordinatorTimedoutTrx = Metrics.meter(coordinator.createMetricName("TimedoutTrx"));
+        coordinatorRecoveryDelay = Metrics.timer(coordinator.createMetricName("RecoveryDelay"));
+        coordinatorRecoveryDuration = Metrics.timer(coordinator.createMetricName("RecoveryTime"));
+        fastPathToTotal = new RatioGaugeSet(coordinatorFastPathTrx, RatioGaugeSet.sum(coordinatorFastPathTrx, coordinatorSlowPathTrx), coordinator, "FastPathToTotal%s");
+    }
+
+    @Override
+    public String toString()
+    {
+        StringBuilder builder = new StringBuilder();
+        builder.append("AccordMetrics [");
+
+        try
+        {
+            for (Field f : getClass().getDeclaredFields())
+            {
+                f.setAccessible(true);
+                if (Counting.class.isAssignableFrom(f.getType()))
+                {
+                    Counting metric = (Counting) f.get(this);
+                    builder.append(String.format("%s: count=%d, ", f.getName(), metric.getCount()));
+                }
+            }
+        }
+        catch (IllegalAccessException e)
+        {
+            throw new RuntimeException(e);
+        }
+        builder.append("]");
+        return builder.toString();
+    }
```

Comment:
Are there any logging bits where we use this `toString()`, or is it mostly here for debugging?

Reply:
It is mostly for debugging, would you like me to change it somehow?

Reply:
As long as it doesn't end up being called on a hot path somewhere, it's probably fine. The alternate would, I guess, be adding the `Counting` instances to a collection and using that in `toString()` (sort of like how we track all metrics in `TableMetrics`), but using reflection isn't necessarily worse for this purpose.